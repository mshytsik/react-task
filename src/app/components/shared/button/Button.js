import { Button, styled } from "@mui/material";

const RedButton = styled(Button)({
  backgroundColor: "#ba0f30",
  borderColor: "#ba0f30",
  "&:hover": {
    backgroundColor: "#d63454",
    borderColor: "#d63454",
  },
});

const StyledButton = ({ callback, children, size = "medium" }) => (
  <RedButton variant="contained" onClick={callback} size={size}>
    {children}
  </RedButton>
);

export default StyledButton;
