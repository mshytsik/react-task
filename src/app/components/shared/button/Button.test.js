import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import Button from "./Button";

const mockCallback = jest.fn(() => "Say hi!");

describe("test Button component", () => {
  test("component is rendered correctly", () => {
    const component = renderer.create(<Button>Click me</Button>).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("callback works correctly", () => {
    render(<Button callback={mockCallback}>Click me</Button>);
    const button = screen.getByText("Click me");
    fireEvent.click(button);
    expect(mockCallback.mock.results[0].value).toBe("Say hi!");
  });

  test("check medium size by default", () => {
    render(<Button>Click me</Button>);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("MuiButton-sizeMedium");
  });
});
