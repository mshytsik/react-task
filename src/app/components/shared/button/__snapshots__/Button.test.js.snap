// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component component is rendered correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-10ni6bz-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Click me
</button>
`;
